### DISCUSSION TITLE
Use wireguard for establishing tunneling between edge router, controller, etc.

### DISCUSSION THREAD

**Discussion by patlegu** [openziti/ziti#1818] [Ideas]
Why not using wireguard for building the encrypted meshed tunnels between components ?

There is a implementation of the product done in Go ==> [Wireguard-go](https://github.com/WireGuard/wireguard-go)

**Comment by oix7** [comment]
What for? It's easier to block and harder to config than modern alternatives available via HTTP3. For example see the complaints/solutions involved here: https://blog.cloudflare.com/zero-trust-warp-with-a-masque/

OpenZiti should continue with its current direction for performance and practical reasons, WireGuard isn't truly needed yet, in my opinion.
